### PR TITLE
Move capability checks to client

### LINKED ISSUE #6315: New navigation is broken when user cannot view reports
**Describe the bug**
When a user does not have the `view_woocommerce_reports` capability, but can `manage_woocommerce` and `edit_shop_orders`, the navigation is broken on the WooCommerce pages they can view.

This is probably due to `view_woocommerce_reports` being somehow tied to all WooCommerce Admin pages/features.

**To Reproduce**

1. Enable the new navigation.
2. Create a user with the above mentioned capabilities. https://wordpress.org/plugins/leira-roles/
3. Visit the orders page `wp-admin/edit.php?post_type=shop_order`
4. Note the broken nav.

**Expected behavior**
Expected the new navigation to still work on these pages.

**Screenshots**
Screen Shot 2021-02-10 at 6 18 25 PM

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #6315

Moves all the privilege checks to client-side and loads WooCommerce Admin for all users.

This adds permission checks to:
* Activity Panel tabs
* WCA Routes
* Menu items

Follow-ups are probably needed to add capability checks for:
* The setup wizard
* Task list
* Inbox notes

### Screenshots

Screen Shot 2021-02-17 at 6 41 44 PM

### Detailed test instructions:

1. Create various non-admin users with custom capabilities.  Make sure to not include the `view_woocommerce_reports` for at least one role.  https://wordpress.org/plugins/leira-roles/
2. Log in as the non-admin users.
3. Check that the correct menu items are shown.
4. Check that there aren't items shown to the user they should not be able to use or interact with.
5. Enable the new navigation under WooCommerce -> Settings -> Advanced -> Features.
6. Check that the users are able to see the new navigation menu.

### CODE REVIEW

File: client/header/activity-panel/index.js [hunk 1]
```diff
@@ -2,17 +2,16 @@
  * External dependencies
```
Comment (becdetat) [inline client/header/activity-panel/index.js:2]: It's probably a good idea to add some smoke test instructions for this after such a big refactor

Comment (joshuatf) [reply client/header/activity-panel/index.js:2]: @becdetat There are testing instructions in the PR and added to `TESTING-INSTRUCTIONS.md`.

I do think that it's worth a good general smoke testing of the app.  Are you suggesting copying over all the generic smoke testing instructions here from the p2 thread or is there something you'd like more specific testing instructions on?

Comment (becdetat) [reply client/header/activity-panel/index.js:2]: I meant testing the activity panel - it doesn't seem to be covered by the test instructions

**Review by becdetat — APPROVED** [review]
Everything tests well, code looks good, I just had the suggestion about adding some general manual test instructions to cover the `activity-panels/index.js` refactor. Pre-approving.

**Comment by joshuatf** [comment]
Thanks for the review, @becdetat. Added a couple lines for smoke testing around the activity panels.

**Pull request merged**